ssl-impl/src/main/java/io/crate/protocols/postgres/ssl/SslConfiguration.java

```diff
@@ -0,0 +1,248 @@
+
+        return new TrustStoreSettings(trustStore, trustManagers);
+    }
+
+    @VisibleForTesting
+    static class TrustStoreSettings {
+
+        KeyStore trustStore;
+        TrustManager[] trustManagers;
+
+        TrustStoreSettings(KeyStore trustStore, TrustManager[] trustManagers) {
+            this.trustStore = trustStore;
+            this.trustManagers = trustManagers;
+        }
+    }
+
+    @VisibleForTesting
+    static KeyStoreSettings loadKeyStore(Settings settings) throws KeyStoreException,
+                                                                   IOException,
+                                                                   NoSuchAlgorithmException,
+                                                                   CertificateException,
+                                                                   UnrecoverableKeyException {
+        String keyStorePath = SslConfigSettings.SSL_KEYSTORE_FILEPATH.setting().get(settings);
+        checkStorePath(keyStorePath, StoreType.key);
+        String keyStorePassword = SslConfigSettings.SSL_KEYSTORE_PASSWORD.setting().get(settings);
+        String keyStoreKeyPassword = SslConfigSettings.SSL_KEYSTORE_KEY_PASSWORD.setting().get(settings);
+        if (keyStoreKeyPassword.isEmpty()) {
+            keyStoreKeyPassword = keyStorePassword;
+        }
+
+        // load the store
+        final KeyStore keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
+        keyStore.load(new FileInputStream(new File(keyStorePath)),
+                      keyStorePassword.isEmpty() ? null : keyStorePassword.toCharArray());
+
+        // initialize a key manager factory with the key store
+        KeyManagerFactory keyFactory = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+        keyFactory.init(
+            keyStore,
+            keyStoreKeyPassword == null || keyStoreKeyPassword.isEmpty() ? null : keyStoreKeyPassword.toCharArray());
+
+        // get the key managers from the factory
+        KeyManager[] keyManagers = keyFactory.getKeyManagers();
+
+        return new KeyStoreSettings(keyStore, keyManagers, keyStorePath, keyStorePassword, keyStoreKeyPassword);
+    }
+
+    @VisibleForTesting
+    static class KeyStoreSettings {
+
+        KeyStore keyStore;
+        KeyManager[] keyManagers;
+        String keyStorePassword;
+        String keyStoreKeyPassword;
+        String keyStorePath;
+
+        KeyStoreSettings(KeyStore keyStore,
+                         KeyManager[] keyManagers,
+                         String keyStorePassword,
+                         String keyStoreKeyPassword,
+                         String keyStorePath) {
+            this.keyStore = keyStore;
+            this.keyManagers = keyManagers;
+            this.keyStorePassword = keyStorePassword;
+            this.keyStoreKeyPassword = keyStoreKeyPassword;
+            this.keyStorePath = keyStorePath;
+        }
+    }
+
+    private enum StoreType {
+        trust,
```

Comment:
I did that on purpose because I use them to construct exception messages, and since this enum is only for that I didn't bother to do something like TRUST("trust"), KEY("key")...